### DISCUSSION TITLE
Upload function not uploading all inputs

### DISCUSSION THREAD

**Discussion by alsonh** [simulot/immich-go#356] [Q&A]
I'm running into an issue with processing a group of three Google Takeout .zip files and not uploading all the files. I've run the same command multiple times, and each time it seems to upload an additional small chunk of files, but always gets to about 51% and then abruptly stops but looks like it has been completed successfully. Anyone run into this issue before? 

The command that I'm running is: immich-go.exe -server [server] -key [key] upload -google-photos -create-albums takeout-*.zip

Some extracts of the log from the three most recent runs

**Run 1**
Input analysis:
---------------
scanned image file                      :   33123
scanned video file                      :    6368
scanned sidecar file                    :   36554
discarded file                          :       6
unsupported file                        :      11
file duplicated in the input            :   17007
associated metadata file                :   39484
missing associated metadata file        :       0

Uploading:
----------
uploaded                                :    3288
upload error                            :       0
file not selected                       :       0
server's asset upgraded with the input  :       7
server has same asset                   :   15873
server has a better asset               :    1683
 | time="2024-07-06T06:32.09 AEST"

**Run 2**
Input analysis:
---------------
scanned image file                      :   33123
scanned video file                      :    6368
scanned sidecar file                    :   36554
discarded file                          :       6
unsupported file                        :      11
file duplicated in the input            :   17000
associated metadata file                :   39484
missing associated metadata file        :       0

Uploading:
----------
uploaded                                :     205
upload error                            :       0
file not selected                       :       0
server's asset upgraded with the input  :       0
server has same asset                   :   18562
server has a better asset               :    1683
 | time="2024-07-06T06:39.38 AEST"

**Run 3**
Input analysis:
---------------
scanned image file                      :   33123
scanned video file                      :    6368
scanned sidecar file                    :   36554
discarded file                          :       6
unsupported file                        :      11
file duplicated in the input            :   17002
associated metadata file                :   39484
missing associated metadata file        :       0

Uploading:
----------
uploaded                                :     202
upload error                            :       0
file not selected                       :       0
server's asset upgraded with the input  :       0
server has same asset                   :   18563
server has a better asset               :    1683
 | time="2024-07-06T10:38.48 AEST"

**Comment by simulot** [comment]
I suggest to look at the log files for having the error cause.

**Reply by alsonh** [reply]
After I added the command -log-level ERROR, but the log files were empty which I presume means there were no errors. Any other suggestions on where to look?

**Reply by simulot** [reply]
I'm not sure the -log-level option works as expected since I have migrated to the structured log package... For some reason, errors aren't logged with the error level... sorry for that

The default option give all you need. Search for "error" in the file

**Reply by alsonh** [reply]
No problems at all. With the default option, there are no errors in the log file. I can send through the log files if you think that would help in any way?

**Reply by simulot** [reply]
You can enable the `-debug-counters`  option. this produce a CSV files with an indication of how each file has been handled.

**Reply by alsonh** [reply]
I tried using that option yesterday and it didn't produce any CSV files in the same directory as the logs. I tried looking for it in the other directories and manually setting it but it still didn't appear

**Reply by simulot** [reply]
Then immich-go crashes before reaching the end... Of course: 51%....
what are the last lines of the log file?

**Reply by alsonh** [reply]
Here's attached the last 100 lines of the most recent log file. 
[log.txt](https://github.com/user-attachments/files/16117657/log.txt)

**Reply by simulot** [reply]
The log shows that the end of the process has been reached the end, without error.
 
I don't understand the "51%" progression

If the server hasn't been updated with another source of photos (via the phone app) and the takeout file is unchanged, there is no reason to get different results

**Reply by alsonh** [reply]
Yeap... it's confused me as well. It hasn't been updated with any other sources or using the app. 

I'll do some more digging and report back if I think of anything else! Thank you!

**Reply by simulot** [reply]
I have fixed an issue in the latest version. Duplicated files weren't detected and counted correctly.

**Reply by alsonh** [reply]
I've just downloaded v0.19.1. It got up to 51% uploading just before it stopped. Will try to make some time tomorrow to have a closer look 
![2024-07-09_22-16-25](https://github.com/simulot/immich-go/assets/3786150/ea358127-9fa5-42bd-a4aa-8a5a2b6ba9a9)

**Reply by simulot** [reply]
Try to enable the `-api-trace` option, the last lines of this big file will which task is hanging

**Reply by alsonh** [reply]
[Uploading immich-go_2024-07-09_22-46-37.trace.log…]()
I had a quick scan through the end and wasn't able to find anything that stood out to me. I'll have a read of the code tomorrow to understand how it works and let you know if I find anything.

**Reply by simulot** [reply]
Unfortunately there is no file attached. You can join my using discord @simulot
